db.go

```diff
@@ -612,17 +617,20 @@ func (db *DB) getMemTables() ([]*memTable, func()) {
 	db.RLock()
 	defer db.RUnlock()
 
-	tables := make([]*memTable, len(db.imm)+1)
```

Comment:
Why do we need this change?

Reply:
In readOnly mode, we should not append the mutable memtable in the tables list. The `+1` is for the `db.mt`.
These `tables` get iterated upon and will cause panic on nil tables. `nil` check at various places can be cumbersome as well as untidy. 
So, either we make this change, or create a slice of size `len(db.imm)` in readOnly mode.

Reply:
okay